subsys/net/lib/http/http_hpack.c

```diff
@@ -0,0 +1,159 @@
+	[HTTP_SERVER_HPACK_CONTENT_ENCODING] = (void *)"content-encoding",
+	[HTTP_SERVER_HPACK_CONTENT_LANGUAGE] = (void *)"content-language",
+	[HTTP_SERVER_HPACK_CONTENT_LENGTH] = (void *)"content-length",
+	[HTTP_SERVER_HPACK_CONTENT_LOCATION] = (void *)"content-location",
+	[HTTP_SERVER_HPACK_CONTENT_RANGE] = (void *)"content-range",
+	[HTTP_SERVER_HPACK_CONTENT_TYPE] = (void *)"content-type",
+	[HTTP_SERVER_HPACK_COOKIE] = (void *)"cookie",
+	[HTTP_SERVER_HPACK_DATE] = (void *)"date",
+	[HTTP_SERVER_HPACK_ETAG] = (void *)"etag",
+	[HTTP_SERVER_HPACK_EXPECT] = (void *)"expect",
+	[HTTP_SERVER_HPACK_EXPIRES] = (void *)"expires",
+	[HTTP_SERVER_HPACK_FROM] = (void *)"from",
+	[HTTP_SERVER_HPACK_HOST] = (void *)"host",
+	[HTTP_SERVER_HPACK_IF_MATCH] = (void *)"if-match",
+	[HTTP_SERVER_HPACK_IF_MODIFIED_SINCE] = (void *)"if-modified-since",
+	[HTTP_SERVER_HPACK_IF_NONE_MATCH] = (void *)"if-none-match",
+	[HTTP_SERVER_HPACK_IF_RANGE] = (void *)"if-range",
+	[HTTP_SERVER_HPACK_IF_UNMODIFIED_SINCE] = (void *)"if-unmodified-since",
+	[HTTP_SERVER_HPACK_LAST_MODIFIED] = (void *)"last-modified",
+	[HTTP_SERVER_HPACK_LINK] = (void *)"link",
+	[HTTP_SERVER_HPACK_LOCATION] = (void *)"location",
+	[HTTP_SERVER_HPACK_MAX_FORWARDS] = (void *)"max-forwards",
+	[HTTP_SERVER_HPACK_PROXY_AUTHENTICATE] = (void *)"proxy-authenticate",
+	[HTTP_SERVER_HPACK_PROXY_AUTHORIZATION] = (void *)"proxy-authorization",
+	[HTTP_SERVER_HPACK_RANGE] = (void *)"range",
+	[HTTP_SERVER_HPACK_REFERER] = (void *)"referer",
+	[HTTP_SERVER_HPACK_REFRESH] = (void *)"refresh",
+	[HTTP_SERVER_HPACK_RETRY_AFTER] = (void *)"retry-after",
+	[HTTP_SERVER_HPACK_SERVER] = (void *)"server",
+	[HTTP_SERVER_HPACK_SET_COOKIE] = (void *)"set-cookie",
+	[HTTP_SERVER_HPACK_STRICT_TRANSPORT_SECURITY] = (void *)"strict-transport-security",
+	[HTTP_SERVER_HPACK_TRANSFER_ENCODING] = (void *)"transfer-encoding",
+	[HTTP_SERVER_HPACK_USER_AGENT] = (void *)"user-agent",
+	[HTTP_SERVER_HPACK_VARY] = (void *)"vary",
+	[HTTP_SERVER_HPACK_VIA] = (void *)"via",
+	[HTTP_SERVER_HPACK_WWW_AUTHENTICATE] = (void *)"www-authenticate",
+};
+
+int http_hpack_table_contains(struct http_client_ctx *ctx, uint32_t key)
+{
+	if (!ctx || !http_hpack_key_is_static((int32_t)key)) {
+		return -EINVAL;
+	}
+
+	if (http_hpack_table_static[key]) {
+		return 0;
+	} else {
+		return -EINVAL;
+	}
+}
+
+int http_hpack_table_get(struct http_client_ctx *ctx, uint32_t key, void **value)
+{
+	if (!ctx || !value || !http_hpack_key_is_static((int32_t)key)) {
+		return -EINVAL;
+	}
+
+	*value = (void *)http_hpack_table_static[key];
+	if (*value) {
+		return 0;
+	} else {
+		return -EINVAL;
+	}
+}
+
+int http_hpack_table_add(struct http_client_ctx *ctx, uint32_t key, void *value)
+{
+	return -ENOSYS;
+}
+
+int http_hpack_table_remove(struct http_client_ctx *ctx, uint32_t key)
+{
+	return -ENOSYS;
+}
+
+char *http_hpack_parse_header(struct http_client_ctx *ctx, int requested_data)
```

Comment:
This may be better to place in `http_parser.c` since it's really just parsing http requests using the "public" http_hpack API